### PR TITLE
Swagger spec for ttl

### PR DESCRIPTION
**Pull request by jamiltz**
@ajres can you take a look at this when you have time (reassign if needed). _exp and show_exp have descriptions but perhaps there are other properties/fields to document.

@snej can you confirm if the TTL spec between SG and CBL's REST APIs are the same.

Connects to #356

### CODE REVIEW

File: swagger/paths/db/doc/doc.yaml [hunk 1]
```diff
@@ -22,9 +22,14 @@ put:
...
   parameters:
-    - $ref: '#/parameters/body'
+    - in: body
+      name: body
+      description: The request body
+      schema:
+        $ref: '#/definitions/Document'
     - $ref: '#/parameters/new_edits'
     - $ref: '#/parameters/rev'
+    - $ref: '#/parameters/show_exp'
```
Comment (adamcfraser) [inline swagger/paths/db/doc/doc.yaml]: show_exp should be on the GET, not the PUT.  The _exp property can be set in the body of the doc.

Comment (jamiltz) [inline swagger/paths/db/doc/doc.yaml]: Ok will fix that. The _exp property is listed in the Document definition in swagger/definitions/index.yaml.

Comment (jamiltz) [inline swagger/paths/db/doc/doc.yaml]: The Document definition is reused across /{db}/_bulk_docs and /{db}/{doc}, that's why it's in a separate file as opposed to inlined.

**Comment by jamiltz** [comment]
Changes per review:
- Move show_exp to GET /{db}/{doc}
- Update _exp description (time format, time interval, couchbase server vs walrus)

File: swagger/definitions/index.yaml [hunk 2]
```diff
@@ -30,6 +30,18 @@ ActiveTasks:
...
+  properties:
+    _id:
+      type: string
+      description: The document ID.
+    _rev:
+      type: string
+      description: The revision number.
+    _exp:
+      type: string
+      description: Expiry time. The time in ISO format at which the document should be removed. For example, the 6th of July 2016 at 17:00 in the BST timezone would be 2016-07-06T17:00:00+01:00. The value can also be an integer that represents the number of seconds from the current time. For Sync Gateway databases, the runtime evaluation of a document expiration time is handled internally by Couchbase Server (TTL is not supported for databases in walrus mode).
```
Comment (adamcfraser) [inline swagger/definitions/index.yaml]: The phrase 'the runtime evaluation of a document expiration time' is probably not that informative to the average user.  I think what we want to communicate is that the following two things happen when you set the _exp in a doc:
- we set the expiry on the Couchbase Server document, to be managed by Couchbase Server
- we record the expiry time in Sync Gateway's metadata, for use by subsequent requests that include the show_exp property

**Comment by jamiltz** [comment]
Updated per last review and rebased.

**Comment by adamcfraser** [comment]
I'm sorry I didn't catch this earlier, but there's some additional options for the _exp value that aren't covered here - it can be set as EITHER an ISO-8601 value, or as a numeric Couchbase Server expiry value.  For the latter case, there are two variations, depending on whether it's less than or greater than 30 days (as described here:
http://docs.couchbase.com/developer/dev-guide-3.0/doc-expiration.html)

Details on the SG handling on the commit:
https://github.com/couchbase/sync_gateway/commit/382149ef755675bded4cc6f14517ea9871a164ae

**Comment by jamiltz** [comment]
Ok, will update.

@snej is the ttl CBL rest API also _exp in PUT db/doc +  /_bulk_docs and show_exp in GET db/doc? Trying to figure out where we can have one definition for both.

**Comment by snej** [comment]
We don't have a REST API for expiration in CBL yet. At least not on iOS/Mac. When we implement it we'll make sure to have it be compatible with the SG one.

**Comment by jamiltz** [comment]
Ok thanks.

**Comment by jamiltz** [comment]
Updated per review.

**Pull request merged**